### ISSUE TITLE
[FR]

### ISSUE THREAD

**Issue by YJonmo** [mlflow/mlflow#8175]
### Willingness to contribute

Yes. I can contribute this feature independently.

### Proposal Summary

Hi there, 

Is there any way to do contour plotting for non numeric hyperparameters using MLFLOW?
Based on what I can see, MLFLOW only creates plots for the numerical parameters which are logged. 
For example, if I want to create a contour plot with the y being my learning rate, x being the type of encoder, and z being the IOU, MLFLOW will not create that as the y axis is nonnumerical.

There are other dashboard toolboxes that do that. For example here is an example from Optuna:

<img width="572" alt="image" src="https://user-images.githubusercontent.com/12040950/230013664-218e8e07-98e9-4afa-8f29-834cb82cb9ae.png">

### Motivation

> #### What is the use case for this feature?

> #### Why is this use case valuable to support for MLflow users in general?

> #### Why is this use case valuable to support for your project(s) or organization?

> #### Why is it currently difficult to achieve this use case?

It just seems to be a missing feature for a great toolbox such as MLFLOW. 

### What component(s) does this bug affect?

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [X] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

### What interface(s) does this bug affect?

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

### What language(s) does this bug affect?

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

### What integration(s) does this bug affect?

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

**Comment by WeichenXu123** [comment]
Thanks for reporting this FR ! It is a very meaningful feature, Exciting to see your PR !

**Comment by YJonmo** [comment]
Hi Weichen,

Just to clarify. I won't be able to create any PR for this anytime soon.
It was more like I was asking if you have such feature.

Regards,
Jacob

**Comment by WeichenXu123** [comment]
> I was asking if you have such feature.

No.

**Comment by B-Step62** [comment]
Closing as not planned: contour plots over non-numeric hyperparameters are a niche visualization that has not gained traction. Exporting runs via `mlflow.search_runs` into a plotting library (or Optuna's own visualizations) covers this today. Feel free to reopen if there is broader demand.
